frontend/common/src/constants/localizedConstants.tsx

```diff
@@ -386,3 +387,70 @@ export const getOperationalRequirement = (
     operationalRequirementId,
     `Invalid Operational Requirement '${operationalRequirementId}'`,
   );
+
+export const provinceOrTerritory = defineMessages({
+  [ProvinceOrTerritory.Alberta]: {
+    defaultMessage: "Alberta",
+    description: "Alberta selection for province or territory input",
+  },
+  [ProvinceOrTerritory.BritishColumbia]: {
+    defaultMessage: "British Columbia",
+    description: "British Columbia selection for province or territory input",
+  },
+  [ProvinceOrTerritory.Manitoba]: {
+    defaultMessage: "Manitoba",
+    description: "Manitoba selection for province or territory input",
+  },
+  [ProvinceOrTerritory.NewfoundlandAndLabrador]: {
+    defaultMessage: "Newfoundland and Labrador",
+    description:
+      "Newfoundland and Labrador selection for province or territory input",
+  },
+  [ProvinceOrTerritory.NewBrunswick]: {
+    defaultMessage: "New Brunswick",
+    description: "New Brunswick selection for province or territory input",
+  },
+  [ProvinceOrTerritory.NorthwestTerritories]: {
+    defaultMessage: "Northwest Territories",
+    description:
+      "Northwest Territories selection for province or territory input",
+  },
+  [ProvinceOrTerritory.NovaScotia]: {
+    defaultMessage: "Nova Scotia",
+    description: "Nova Scotia selection for province or territory input",
+  },
+  [ProvinceOrTerritory.Nunavut]: {
+    defaultMessage: "Nunavut",
+    description: "Nunavut selection for province or territory input",
+  },
+  [ProvinceOrTerritory.Ontario]: {
+    defaultMessage: "Ontario",
+    description: "Ontario selection for province or territory input",
+  },
+  [ProvinceOrTerritory.PrinceEdwardIsland]: {
+    defaultMessage: "Prince Edward Island",
+    description:
+      "Prince Edward Island selection for province or territory input",
+  },
+  [ProvinceOrTerritory.Quebec]: {
+    defaultMessage: "Quebec",
+    description: "Quebec selection for province or territory input",
+  },
+  [ProvinceOrTerritory.Saskatchewan]: {
+    defaultMessage: "Saskatchewan",
+    description: "Saskatchewan selection for province or territory input",
+  },
+  [ProvinceOrTerritory.Yukon]: {
+    defaultMessage: "Yukon",
+    description: "Yukon selection for province or territory input",
+  },
+});
+
+export const getProvinceOrTerritory = (
+  provinceOrTerritoryId: string | number,
+): MessageDescriptor =>
+  getOrThrowError(
+    provinceOrTerritory,
+    provinceOrTerritoryId,
+    `Invalid Operational Requirement '${provinceOrTerritoryId}'`,
```

Comment:
Good catch, this would be the result of copy + paste with not enough oversight on my part.

Resolved with [fix typo in get province error message](https://github.com/GCTC-NTGC/gc-digital-talent/pull/2413/commits/84c5afdc3ba580ea1788baba1937d4622e36e5bf)